src/observable/GenerateObservable.ts

```diff
@@ -0,0 +1,129 @@
+import {Observable} from '../Observable' ;
+import {Scheduler} from '../Scheduler';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+import {Action} from '../scheduler/Action';
+
+import {isScheduler} from '../util/isScheduler';
+
+const selfSelector = <T>(value: T) => value;
+
+export type ConditionFunc<S> = (state: S) => boolean;
+export type IterateFunc<S> = (state: S) => S;
+export type ResultFunc<S, T> = (state: S) => T;
+
+interface SchedulerState<T, S> {
+  state: S;
+  params: {
+    subscriber: Subscriber<T>;
+    condition?: ConditionFunc<S>;
+    iterate: IterateFunc<S>;
+    resultSelector: ResultFunc<S, T>;
+  };
+}
```

Comment:
what about flatten `SchedulerState` similar to

```js
SchedulerState<T, S> {
  state: S;
  subscriber: Subscriber<T>;
  condition?: ConditionFunc<S>;
  iterate: IterateFunc<S>;
  resultSelector: ResultFunc<S, T>;
}
```
? then pass mutated object at line 127.